ImageD11/peakselect.py

```diff
@@ -73,17 +73,73 @@ def rings_mask(cf, dstol, dsmax, cell=None):
             m |= (abs(cf.ds - v) < dstol)
     return m
 
-
+# separated plot logic and core computation logic,
 def sorted_peak_intensity_mask(colf, uself=True, frac=0.995, B=0.2, doplot=None):
-    """Create a boolean mask for a columnfile
-       Based on peaks sorted by fractional intensity
-       Keeps peaks[:frac] where peaks are sorted by intensity"""
-    # correct intensities for structure factor (decreases with 2theta)
+    """
+    Create a boolean mask for a columnfile based on peaks sorted by fractional intensity.
+
+    Args:
+        colf: Input column file object.
+        uself: Apply Lorentz factor correction (default: True).
+        frac: Fraction of peaks to keep based on intensity.
+        B: Thermal factor for intensity correction.
+        doplot: Optional plotting argument.
+
+    Returns:
+        mask: Boolean mask for selected peaks.
+    """
```

Comment:
@jonwright @jadball 

Do you follow a standard for docstrings ? Is it okay to format them like that ?